finality-aleph/src/metrics.rs

```diff
@@ -71,11 +71,26 @@ pub(crate) enum Checkpoint {
 
 #[derive(Clone)]
 pub struct Metrics<H: Key> {
-    inner: Arc<Mutex<Inner<H>>>,
+    inner: Option<Arc<Mutex<Inner<H>>>>,
 }
 
 impl<H: Key> Metrics<H> {
-    pub fn register(registry: &Registry) -> Result<Self, PrometheusError> {
+    pub fn register<F: FnOnce(PrometheusError)>(
+        registry: Option<&Registry>,
+        on_registration_failed: F,
+    ) -> Metrics<H> {
+        let inner = match registry {
+            Some(r) => Self::try_register(r).map_err(on_registration_failed).ok(),
+            None => {
+                on_registration_failed(PrometheusError::Msg("registry not available".to_string()));
```

Comment:
here you could log that we run without metrics enabled (most likely warn level)

Reply:
Just to make sure - logging on unit tests does nothing serious, right? If so, I think we can move error handling entirely to that function.

Reply:
well, unit test can show logs if invoked with flag `-- --no-capture` so imho its good to have them anyway